cmd/provision/main.go

```diff
@@ -30,58 +32,70 @@ const (
+	defMfCertsURL      = "http://localhost/certs"
 	defProvisionCerts  = "false"
 	defProvisionBS     = "true"
 	defBSAutoWhitelist = "true"
 	defBSContent       = `{}`
 
-	envClientTLS        = "MF_ENV_CLIENTS_TLS"
+	envConfigFile       = "MF_PROVISION_CONFIG_FILE"
+	envLogLevel         = "MF_PROVISION_LOG_LEVEL"
+	envHTTPPort         = "MF_PROVISION_HTTP_PORT"
+	envTLS              = "MF_PROVISION_ENV_CLIENTS_TLS"
 	envCACerts          = "MF_PROVISION_CA_CERTS"
 	envServerCert       = "MF_PROVISION_SERVER_CERT"
 	envServerKey        = "MF_PROVISION_SERVER_KEY"
-	envMQTTURL          = "MF_MQTT_URL"
-	envUsersLocation    = "MF_USERS_LOCATION"
-	envThingsLocation   = "MF_THINGS_LOCATION"
-	envLogLevel         = "MF_PROVISION_LOG_LEVEL"
-	envHTTPPort         = "MF_PROVISION_HTTP_PORT"
-	envMfUser           = "MF_USER"
-	envMfPass           = "MF_PASS"
-	envThingIDs         = "MF_THING_IDS"
-	envMfBSURL          = "MF_BS_SVC_URL"
-	envMfBSWhiteListURL = "MF_BS_SVC_WHITELISTE_URL"
-	envMfCertsURL       = "MF_CERTS_SVC_URL"
-	envProvisionCerts   = "MF_X509_PROVISIONING"
-	envProvisionBS      = "MF_BS_CONFIG_PROVISIONING"
-	envBSAutoWhiteList  = "MF_BS_AUTO_WHITELIST"
-	envBSContent        = "MF_BS_CONTENT"
+	envMQTTURL          = "MF_PROVISION_MQTT_URL"
+	envUsersLocation    = "MF_PROVISION_USERS_LOCATION"
+	envThingsLocation   = "MF_PROVISION_THINGS_LOCATION"
+	envMfUser           = "MF_PROVISION_USER"
+	envMfPass           = "MF_PROVISION_PASS"
+	envMfApiKey         = "MF_PROVISION_API_KEY"
+	envMfCertsURL       = "MF_PROVISION_CERTS_SVC_URL"
+	envProvisionCerts   = "MF_PROVISION_X509_PROVISIONING"
+	envMfBSURL          = "MF_PROVISION_BS_SVC_URL"
+	envMfBSWhiteListURL = "MF_PROVISION_BS_SVC_WHITELIST_URL"
+	envProvisionBS      = "MF_PROVISION_BS_CONFIG_PROVISIONING"
+	envBSAutoWhiteList  = "MF_PROVISION_BS_AUTO_WHITELIST"
+	envBSContent        = "MF_PROVISION_BS_CONTENT"
 )
 
-type config struct {
-	provision.Config
-	logLevel         string
-	clientTLS        bool
-	caCerts          string
-	serverCert       string
-	serverKey        string
-	httpPort         string
-	mfBSURL          string
-	mfCertsURL       string
-	mfBSWhiteListURL string
-}
+var (
+	errMissingConfigFile        = errors.New("missing config file setting")
+	errFailedToGetAutoWhiteList = errors.New("failed to get auto whitelist setting")
+	errFailGettingCertSettings  = errors.New("failed to get certificate file setting")
+	errFailGettingTLSConf       = errors.New("failed to get TLS setting")
+	errFailGettingProvBS        = errors.New("failed to get BS url setting")
+)
 
 func main() {
-	cfg := loadConfig()
-
-	logger, err := logger.New(os.Stdout, cfg.logLevel)
+	cfg, err := loadConfig()
 	if err != nil {
 		log.Fatalf(err.Error())
 	}
+	logger, err := logger.New(os.Stdout, cfg.Server.LogLevel)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	cfg, err = loadConfigFromFile(cfg, logger)
+	if err != nil {
+		logger.Warn(fmt.Sprintf("Continue with settings from env vars: %s", err))
```

Comment:
Use something like `failed to load config from file %s`

Reply:
If i write failed that sounds like a problem, this is like a warning and service can continue to work as it is intented

Reply:
Warning flag is enough to understand that it's not an error. My remark was about the text itself which is confusing.

Reply:
@mteodor maybe a compromise like: "Failed to load config file. Using ENV vars and deafault values."